ocs_ci/ocs/machine.py

```diff
@@ -445,6 +445,105 @@ def create_custom_machineset(
+                machine.get("spec")
+                .get("template")
+                .get("spec")
+                .get("providerSpec")
+                .get("value")
+                .get("template_name")
+            )
+            cls_id = (
+                machine.get("spec")
+                .get("selector")
+                .get("matchLabels")
+                .get("machine.openshift.io/cluster-api-cluster")
+            )
+            socket = (
+                machine.get("spec")
+                .get("template")
+                .get("spec")
+                .get("providerSpec")
+                .get("value")
+                .get("cpu")
+                .get('sockets')
+            )
+
+            machineset_yaml = templating.load_yaml(constants.MACHINESET_YAML_RHV)
+
+            # Update machineset_yaml with required values.
+            machineset_yaml["metadata"]["labels"][
+                "machine.openshift.io/cluster-api-cluster"
+            ] = cls_id
+            machineset_yaml["metadata"]["name"] = f"{cls_id}-{role}-{zone}"
+            machineset_yaml["spec"]["selector"]["matchLabels"][
+                "machine.openshift.io/cluster-api-cluster"
+            ] = cls_id
+            machineset_yaml["spec"]["selector"]["matchLabels"][
+                "machine.openshift.io/cluster-api-machineset"
+            ] = f"{cls_id}-{role}-{zone}"
+            machineset_yaml["spec"]["template"]["metadata"]["labels"][
+                "machine.openshift.io/cluster-api-cluster"
+            ] = cls_id
+            machineset_yaml["spec"]["template"]["metadata"]["labels"][
+                "machine.openshift.io/cluster-api-machine-role"
+            ] = role
+            machineset_yaml["spec"]["template"]["metadata"]["labels"][
+                "machine.openshift.io/cluster-api-machine-type"
+            ] = role
+            machineset_yaml["spec"]["template"]["metadata"]["labels"][
+                "machine.openshift.io/cluster-api-machineset"
+            ] = f"{cls_id}-{role}-{zone}"
+            machineset_yaml["spec"]["template"]["spec"]["providerSpec"]["value"][
+                "cluster_id"
+            ] = cls_uuid
+            machineset_yaml["spec"]["template"]["spec"]["providerSpec"]["value"][
+                "template_name"
+            ] = template_name
+            machineset_yaml["spec"]["template"]['spec']["providerSpec"]["value"][
+                "cpu"
+            ]["sockets"] = socket
+
+            # Apply the labels
+            if labels:
+                for label in labels:
+                    machineset_yaml["spec"]["template"]["spec"]["metadata"]["labels"][
+                        label[0]
+                    ] = label[1]
+                # Remove app label in case of infra nodes
+                if role == "infra":
+                    machineset_yaml["spec"]["template"]["spec"]["metadata"][
+                        "labels"
+                    ].pop(constants.APP_LABEL, None)
```

Comment:
can lines 524 - 533  be written as a generic function?

Reply:
create_custom_machineset() has above lines twice, planning to take update/enhancement in 4.9 time.
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/machine.py#L183
Scale backlog JIRA card https://projects.engineering.redhat.com/browse/OCSQE-657